README.md

```diff
@@ -136,122 +171,279 @@ Every resource in this repository is essential for the overall template setup. N
-    ]
-  }
+  const cssLinks = [
+    'https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css',
+    'style.css',
+  ];
+
+  // Notice that root component is the `File` component.
+  return (
+    <File name="index.html">
+      <HTML>
+        <Head cssLinks={cssLinks} />
+        <Body>
+          <BodyContent asyncapi={asyncapi} />
+          ...
+        </Body>
+      </HTML>
+    </File>
+  );
+}
+
+// Custom component inside template file
+function BodyContent({ asyncapi }) {
+  // implemntation...
 }
 ```
 
-Notice that you can specify one or many hooks you want to reuse from the library instead of all the hooks. In this Template, we use `createAsyncapiFile` responsible for creating the `asyncapi.yaml` file in the directory where template files get generated. This hook also supports custom parameter that I can specify in my configuration:
-```json
-"generator": {
-  "parameters": {
-    "asyncapiFileDir": {
-      "description": "Custom location of the AsyncAPI file that you provided as an input into generation. By default, it is located in the root of the output directory."
-    }
+Each custom component must returns as output pure `string`, another custom component, `null` or `undefined`. Nothing will be rendered for the last two.
+
+Recommended place to create reusable chunks is the `partials` folder. The reusable parts can be located both in the `template` folder as in another named folder. The only exception is the `hooks` folder, it is reserved for the Generator.
+
+#### Using JS in template
+
+When you use React, you are actually using JS, so you can apply conditions to rendering, split functionality into separate/reusable functions, create compisition, extend/mix functions etc.
+
+```js
+/* 
+ * If asyncapi has `externalDocs` property then the Generator will return appropriate string,
+ * otherwise won't render anything.
+ */
+function ExternalDocs({ asyncapi }) {
+  if (!asyncapi.hasExternalDocs()) return null;
+  return `Don't forget to visit our website ${asyncapi.externalDocs().url()}.`
+}
+```
+
+#### Retrieve rendered content from children
+
+Each component has a `childrenContent` property. It is the processed children content of a component into a pure string. You can use it for compositions in your component.
+
+```js
+function CustomComponent({ childrenContent }) {
+  return `some text at the beginning: ${childrenContent}`
+}
+
+function RootComponent() {
+  return (
+    <CustomComponent>
+      some text at the end.
+    </CustomComponent>
+  );
+}
+```
+
+Then output from `RootComponent` will be `some text at the beginning: some text at the end.`.
+
+#### Render component to string
+
+If you need process the React component to string you should use `render` function from `@asyncapi/generator-react-sdk` package. This function transforms given component (and its children) and returns pure string. You can check [template/index.js](template/index.js) to see how it could by done:
```

Comment:
```suggestion
If you need to process the React component to string you should use `render` function from `@asyncapi/generator-react-sdk` package. This function transforms a given component (and its children) and returns the pure string representation. You can check the [template/index.js](template/index.js) to see how it could by done:
```